estate/models/estate_property.py

```diff
@@ -0,0 +1,111 @@
+from odoo import models, fields, api
+from dateutil.relativedelta import relativedelta
+
+
+class EstateProperty(models.Model):
+    _name = "estate.property"
+    _description = "estate.property"
+
+    name = fields.Char(string="Title", required=True)
+    description = fields.Text(string="Description")
+    postcode = fields.Char(string="Postcode")
+    date_availability = fields.Date(
+        string="Availability From",
+        copy=False,
+        default=lambda self: fields.Date.today() + relativedelta(months=3),
+    )
+    expected_price = fields.Float(string="Expected Price", required=True)
+    selling_price = fields.Float(
+        string="Selling Price",
+        readonly=True,
+        copy=False,
+    )
+    best_price = fields.Float(
+        string="Best Price",
+        compute="_compute_best_price",
+        store=True,
+    )
+    property_type_id = fields.Many2one(
+        string="Property Type",
+        comodel_name="estate.property.type",
+    )
+    tag_ids = fields.Many2many(
+        string="Tags",
+        comodel_name="estate.property.tag",
+    )
+    buyer_id = fields.Many2one(string="Buyer", comodel_name="res.partner")
+    seller_id = fields.Many2one(
+        string="Seller",
+        comodel_name="res.users",
+        default=lambda self: self.env.user,
+    )
+    bedrooms = fields.Integer(string="Bedrooms", default=2)
+    living_area = fields.Integer(string="Living Area (sqm)")
+    facades = fields.Integer(string="Facades")
+    garage = fields.Boolean(string="Garage")
+    garden = fields.Boolean(string="Garden")
+    garden_area = fields.Integer(string="Garden Area")
+    garden_orientation = fields.Selection(
+        [
+            ("north", "North"),
+            ("south", "South"),
+            ("east", "East"),
+            ("west", "West"),
+        ],
+        string="Garden Orientation",
+    )
+    total_area = fields.Integer(
+        string="Total Area (sqm)",
+        compute="_compute_total_area",
+        store=True,
+    )
+    state = fields.Selection(
+        [
+            ("new", "New"),
+            ("offer_received", "Offer Received"),
+            ("offer_accepted", "Offer Accepted"),
+            ("sold", "Sold"),
+            ("cancelled", "Cancelled"),
+        ],
+        string="State",
+        default="new",
+        required=True,
+        copy=False,
+    )
+    offer_ids = fields.One2many(
+        string="Offers",
+        comodel_name="estate.property.offer",
+        inverse_name="property_id",
+    )
+    active = fields.Boolean(string="Active", default=True)
+
+    # ================================================================== #
+    #                              COMPUTES                              #
+    # ================================================================== #
```

Comment:
```suggestion
    # === Computes ===
```
Seems a bit overkilled. Small separation should be enough here.
Up to you